packages/KDateTimePicker/KDateTimePicker.vue

```diff
@@ -0,0 +1,632 @@
+     * back to the parent.
+     */
+    const clearSelection = () => {
+      selectedCalendarRange.value = null
+      abbreviatedDisplay.value = props.placeholder
+      fullRangeDisplay.value = ''
+      selectedRange.value = { start: '', end: '', timePeriodsKey: '' }
+
+      if (hasTimePeriods) {
+        selectedTimeframe.value = props.timePeriods[0]
+      }
+
+      // If the calendar has focus, and `range` is set to false, we are displaying
+      // a single date / time picker; therefore, need to emit an empty string.
+      //
+      // Else, we are displaying a date / time range (relative timeframes or calendar range)
+      if (!props.range && showCalendar) {
+        emit('changed', '')
+      } else {
+        emit('changed', selectedRange.value)
+      }
+    }
+
+    /**
+     * Displays selected date/time/range as a human readable string.
+     * The date formatting string is dynamically determined based on
+     * the current mode of the instance (Custom vs Relative)
+     * @param {*} range A set of `start` and `end` Unix timestamps
+     */
+    const formatDisplayDate = (range) => {
+      const { start, end } = range
+      let fmtStr = 'PP'
+
+      // Determines the human timestamp readout format string; subject to change
+      if (!hasCalendar.value && hasTimePeriods.value) {
+        fmtStr = 'PP hh:mm a'
+      } else if (props.mode === 'date') {
+        fmtStr = 'PP'
+      } else if (props.mode === 'time') {
+        fmtStr = 'PP hh:mm a'
+      } else if (props.mode === 'dateTime') {
+        fmtStr = 'PP hh:mm a'
+      }
+
+      // Determine whether an end date/time should be displayed in readout
+      return range.end && (props.range || hasTimePeriods.value)
+        ? `${format(start, fmtStr)} - ${format(end, fmtStr)}`
+        : `${format(start, fmtStr)}`
+    }
+
+    /**
+     * Formats user-facing display text when a new timeframe is selected.
+     * @param {*} tf Timeframe object
+     */
+    const getTimeframeText = (tf) => {
+      return `${tf.prefix} ${tf.timeframeText}`
+    }
+
+    /**
+     * Once a selection is made, the value is emitted back to the parent,
+     * and the input field value is updated as a visual confirmation.
+     */
+    const submitTimeFrame = async () => {
+      if (props.range || hasTimePeriods.value) {
+        emit('changed', selectedRange.value)
+
+        // Determine which tab has focus, then update input field text
+        if (showCalendar && selectedRange.value) {
+          abbreviatedDisplay.value = formatDisplayDate(selectedRange.value)
+        } else {
+          abbreviatedDisplay.value = getTimeframeText(selectedTimeframe.value)
+        }
+      } else {
+        // Update input field text for single date / time instance
+        abbreviatedDisplay.value = formatDisplayDate(selectedRange.value)
+        emit('changed', new Date(selectedRange.value.start))
+      }
+
+      await nextTick(() => {
+        hidePopover.value = true
+      })
```

Comment:
Could you just call the `handleClose` function here instead?

Reply:
Hah. Good call :)